uqcsbot/api.py

```diff
@@ -1,5 +1,7 @@
-from typing import List
+from typing import List, Union, Awaitable
+from functools import partial
 from slackclient import SlackClient
+import asyncio
 
 
 # see https://api.slack.com/docs/pagination for details
```

Comment:
For some reason github can't let me comment on it, but can you update the paginate_api_call while you are here?

Reply:
Or at least somehow move pagination into the API Wrapper.

Reply:
Made a `Paginator` class that does iteration (async or otherwise) which is cool